commit 0455e72c9ae9b7e9589f2cc5ba5bc7804be83342
Author: Eric Auger <eric.auger@redhat.com>
Date:   Fri Jul 22 16:20:38 2016 +0000

    KVM: Add devid in kvm_kernel_irq_routing_entry
    
    Enhance kvm_kernel_irq_routing_entry to transport the device id
    field, devid. A new flags field makes possible to indicate the
    devid is valid. Those additions are used for ARM GICv3 ITS MSI
    injection. The original struct msi_msg msi field is replaced by
    a new custom structure that embeds the new fields.
    
    Signed-off-by: Eric Auger <eric.auger@redhat.com>
    Suggested-by: Radim Krčmář <rkrcmar@redhat.com>
    Acked-by: Radim Krčmář <rkrcmar@redhat.com>
    Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>

diff --git a/include/linux/kvm_host.h b/include/linux/kvm_host.h
index 614a98137c5f..a15828fe845c 100644
--- a/include/linux/kvm_host.h
+++ b/include/linux/kvm_host.h
@@ -317,7 +317,13 @@ struct kvm_kernel_irq_routing_entry {
 			unsigned irqchip;
 			unsigned pin;
 		} irqchip;
-		struct msi_msg msi;
+		struct {
+			u32 address_lo;
+			u32 address_hi;
+			u32 data;
+			u32 flags;
+			u32 devid;
+		} msi;
 		struct kvm_s390_adapter_int adapter;
 		struct kvm_hv_sint hv_sint;
 	};